plugins/outputs/otlp/otlp.go

```diff
@@ -0,0 +1,379 @@
+package otlp
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"sort"
+	"time"
+
+	// TODO: this should be simplified
+	// Imports the OTLP client from the prometheus sidecar
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	metricsService "go.opentelemetry.io/proto/otlp/collector/metrics/v1"
+	otlpcommonpb "go.opentelemetry.io/proto/otlp/common/v1"
+	otlpmetricpb "go.opentelemetry.io/proto/otlp/metrics/v1"
+	otlpresourcepb "go.opentelemetry.io/proto/otlp/resource/v1"
+	"google.golang.org/grpc/metadata"
+)
+
+// OTLP is the OpenTelemetry Protocol config info.
+type OTLP struct {
+	Endpoint   string            `toml:"endpoint"`
+	Timeout    string            `toml:"timeout"`
+	Headers    map[string]string `toml:"headers"`
+	Attributes map[string]string `toml:"attributes"`
+
+	Namespace string
+	Log       telegraf.Logger
+
+	client       *Client
+	resourceTags []*telegraf.Tag
+	grpcTimeout  time.Duration
+}
+
+const (
+	// QuotaLabelsPerMetricDescriptor is the limit
+	// to labels (tags) per metric descriptor.
+	QuotaLabelsPerMetricDescriptor = 10
+	// QuotaStringLengthForLabelKey is the limit
+	// to string length for label key.
+	QuotaStringLengthForLabelKey = 100
+	// QuotaStringLengthForLabelValue is the limit
+	// to string length for label value.
+	QuotaStringLengthForLabelValue = 1024
+
+	// StartTime for cumulative metrics.
+	StartTime = int64(1)
+	// MaxInt is the max int64 value.
+	MaxInt = int(^uint(0) >> 1)
```

Comment:
Do we need those exported?

Reply:
They do not, I've removed the unused variables and those left are no longer exported